airflow/models/dag.py

```diff
@@ -1025,11 +1025,68 @@ def get_task_instances(self, start_date=None, end_date=None, state=None, session
             start_date = (timezone.utcnow() - timedelta(30)).date()
             start_date = timezone.make_aware(datetime.combine(start_date, datetime.min.time()))
 
-        tis = session.query(TaskInstance).filter(
-            TaskInstance.dag_id == self.dag_id,
-            TaskInstance.execution_date >= start_date,
-            TaskInstance.task_id.in_([t.task_id for t in self.tasks]),
+        tis = self._get_task_instances(
+            task_ids=None,
+            start_date=start_date,
+            end_date=end_date,
+            state=state,
+            include_subdags=False,
+            include_parentdag=False,
+            include_dependent_dags=False,
+            as_pk_tuple=False,
+            session=session,
         )
+        tis = tis.order_by(TaskInstance.execution_date).all()
+        return tis
+
+    def _get_task_instances(
+        self,
+        *,
+        task_ids,
+        start_date,
+        end_date,
+        state,
+        include_subdags,
+        include_parentdag,
+        include_dependent_dags,
+        as_pk_tuple,
+        session,
+        dag_bag=None,
+        recursion_depth=0,
+        max_recursion_depth=None,
+        visited_external_tis=None,
+    ):
```

Comment:
Might be nice to have type annotations here.

Reply:
Done